packages/stack/src/views/Stack/CardContainer.tsx

```diff
@@ -152,8 +165,19 @@ function CardContainer({
     } else {
       onPageChangeCancel?.();
     }
-
-    onTransitionStart?.({ route }, closing);
+    /*fix for https://github.com/react-navigation/react-navigation/issues/9870
+       We trigger transition events only on the incoming (current) screen
```

Comment:
"current screen" doesn't mean incoming screen, it means the screen where you add the listener - that is animating/transitioning.

Reply:
Thanks for clarification. I will update the docs.

Reply:
Since both the current / previous screen (can) animate in a transition (cardStyleInterpolator supports this) . Was it intentended to send events to both screen assuming they had listeners?